dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ResourcesV2Controller.java

```diff
@@ -0,0 +1,802 @@
+import org.apache.dolphinscheduler.api.utils.PageInfo;
+import org.apache.dolphinscheduler.api.utils.Result;
+import org.apache.dolphinscheduler.common.Constants;
+import org.apache.dolphinscheduler.common.enums.ProgramType;
+import org.apache.dolphinscheduler.common.enums.UdfType;
+import org.apache.dolphinscheduler.common.utils.ParameterUtils;
+import org.apache.dolphinscheduler.dao.entity.Resource;
+import org.apache.dolphinscheduler.dao.entity.UdfFunc;
+import org.apache.dolphinscheduler.dao.entity.User;
+import org.apache.dolphinscheduler.spi.enums.ResourceType;
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestAttribute;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.multipart.MultipartFile;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiImplicitParam;
+import io.swagger.annotations.ApiImplicitParams;
+import io.swagger.annotations.ApiOperation;
+import springfox.documentation.annotations.ApiIgnore;
+
+
+
+/**
+ * resources controller
+ */
+@Api(tags = "RESOURCES_TAG")
+@RestController
+@RequestMapping("/v2/resources")
+public class ResourcesV2Controller extends BaseController {
+
+    private static final Logger logger = LoggerFactory.getLogger(ResourcesV2Controller.class);
+
+    @Autowired
+    private ResourcesService resourceService;
+    @Autowired
+    private UdfFuncService udfFuncService;
+
+    /**
+     * create resource director
+     *
+     * @param loginUser        login user
+     * @param directoryRequest create directory request
+     * @return create result code
+     */
+    @ApiOperation(value = "createDirectory",
+        notes = "CREATE_RESOURCE_NOTES",
+        consumes = "application/json",
+        httpMethod = "POST")
+    @PostMapping(value = "/directory")
+    @ApiException(CREATE_RESOURCE_ERROR)
+    @AccessLogAnnotation(ignoreRequestArgs = "loginUser")
+    public Result<Resource> createDirectory(@ApiIgnore @RequestAttribute(value = Constants.SESSION_USER) User loginUser,
```

Comment:
i think Result is a common result type ，so i use i with a pojo generic-type .do you means don't use Result ?

Reply:
Yes.

Reply:
Sorry i am puzzled，if we don't use Result ，it will increase large POJO  ，and  some api just need a simple result such as  `{"code": 0 ,"msg":""}` , and how to keep the format of response the same ， do you have a success PR for reference？

Reply:
You can take a look at #10290.

Reply:
> You can take a look at #10290.

Get

Reply:
> You can take a look at #10290.

I view AccessTokenV2Controller ,the class CreateTokenResponse extend Result ,but  accessTokenService is also use Result , so can i just wrap the Response like that？

Reply:
> I view AccessTokenV2Controller ,the class CreateTokenResponse extend Result ,but accessTokenService is also use Result , so can i just wrap the Response like that？

Yes. In order to be compatible with old and new apis and reduce modification costs.